yt_dlp/extractor/crunchyroll.py

```diff
@@ -97,6 +106,7 @@ def _update_auth(self):
                     'and your browser\'s User-Agent (with --user-agent)', expected=True)
             raise
 
+        CrunchyrollBaseIE._ACCESS_TOKEN = auth_response['access_token']
```

Comment:
Why save the access token and not decode the token and set `self.is_premium` here immediately?